### ISSUE TITLE
Unexpected type when querying referenced table with count

### ISSUE THREAD

**Issue by peperoli** [supabase/supabase-js#939]
# Bug report

## Describe the bug

When querying a referenced table with `count` option like this:

```ts
const supabase = createClientComponentClient<Database>()

const { data } = supabase.from('concerts').select(`*, bands_count:j_concert_bands(count)`
```

`count` is returned correctly:

```JSON
[
  {
    "id": 1,
    "bands_count": [
      {
        "count": 29
      }
    ]
  }
]
```

But when using the returned `count` value there's a type error:

```ts
data.forEach(item => {
  const count = item.bands_count[0].count
  // Property 'count' does not exist on type 'SelectQueryError<"Referencing missing column `count`">'.
}
```

## Expected behavior

No error message because `count` does in fact exist.

## System information

- Version of supabase-js: 2.39.0
- Version of Node.js: 18.19.0

**Comment by salvinoto** [comment]
Im having same issue as well

**Comment by peperoli** [comment]
This issue is active in the postgrest-js repo: https://github.com/supabase/postgrest-js/issues/447

**Comment by avallete** [comment]
A fix has been released in `supabase-js` v2.46.0. I'm closing this issue, but feel free to reopen if you encounter any further errors.
